_rules/printable-characters-shortcut-ffbc54.md

```diff
@@ -0,0 +1,363 @@
+---
+id: ffbc54
+name: No keyboard shortcut uses only printable characters
+rule_type: atomic
+description: |
+  This rule checks that if keyboard shortcuts are implemented using only printable characters, then there is a mechanism to disable the shortcut, or to remap the shortcut to use one or more non-printable characters keys, or the shortcut for a user interface component is only available when that component has focus.
+accessibility_requirements:
+  wcag21:2.1.4: # Character Key Shortcuts (A)
+    forConformance: true
+    failed: not satisfied
+    passed: satisfied
+    inapplicable: further testing needed
+input_aspects:
+  - DOM Tree
+acknowledgments:
+  authors:
+    - João Vicente
+    - Carlos Duarte
+---
+
+## Applicability
+
+The rule applies to any [keyboard event][] for which all of the following is true: 
+- the event's attribute `key` is a [printable character][] key; and
+- the event's method `getModifierState` returns `false` for any of the [valid modifier keys][]; and
+- the event causes [changes in the content][changes in content] of the [HTML document][].
+
+## Expectation
+
+For each test target at least one of the following is true:
+- (**Remap**:) there is at least one [set of clearly labeled instruments][] to [block events][blocked event] that use the [same key][same key events] as the test target and whose `getModifierState` method returns `false` for any of the [valid modifier keys][]; or
+- (**Focus**:) the [event target][] has a [semantic role][] that inherits from the [abstract role](https://www.w3.org/TR/wai-aria/#abstract_roles) of `widget`.
```

Comment:
I don't think this is the same as the element only reacting to keyboard shortcuts if it has focus. For example, passed example 5 would fail right? Because the event target doesn't have a widget role. Or do I misunderstand event target?

Reply:
You're correct. This is not the same. But the current rule formulation does not target "keyboard shortcuts", but "keyboard events". By definition of keyboard event, it is always dispatched to the element that has focus. Therefore, we cannot check if the element has focus (it always has).
Because the SC requires the focus condition only for "user interface components", we used this workaround. Instead of checking if the element has focus, we check if the element with focus is a widget (that is, a user interface component).

In passed example 5, the event target that requires focus for the shortcut to work is the `input type="text` element. Therefore, it inherits from the `widget` role. All other elements make the event not applicable because it won't cause changes in content.